### PR TITLE
Remap Ids in SC internal, unordered containers.

### PR DESCRIPTION
**Pull request by carlitosan**
fix up saved data that WAS stored in an unordered container after all

Signed-off-by: carlitosan <82187351+carlitosan@users.noreply.github.com>

### CODE REVIEW

**Review by lumberyard-employee-dm — APPROVED** [review]

File: Gems/ScriptCanvas/Code/Editor/Assets/ScriptCanvasFileHandling.cpp [hunk 1]
```diff
@@ -77,6 +77,22 @@ namespace ScriptCanvasFileHandlingCpp
...
+    //
+    // EntityIds contained in variable (those set to self or the graph unique id, will be ignored)
+    void MakeGraphComponentEntityIdsUnique(AZ::Entity* entity, AZ::SerializeContext* serializeContext)
+    {
+        AZStd::unordered_map<AZ::EntityId, AZ::EntityId> remappedIds;
+        remappedIds[AZ::EntityId()] = AZ::EntityId();
+        remappedIds[ScriptCanvas::GraphOwnerId] = ScriptCanvas::GraphOwnerId;
+        remappedIds[ScriptCanvas::UniqueId] = ScriptCanvas::UniqueId;
+
+        AZ::IdUtils::Remapper<AZ::EntityId>::GenerateNewIdsAndFixRefs(entity, remappedIds, serializeContext);
```
Comment (onecent1101) [inline Gems/ScriptCanvas/Code/Editor/Assets/ScriptCanvasFileHandling.cpp:94]: can you explain what is difference between enumerating objects from serialize context and this method?

**Review by onecent1101 — APPROVED** [review]
LG

File: Gems/ScriptCanvas/Code/Editor/Assets/ScriptCanvasFileHandling.cpp (hunk 1 above)
Comment (lsemp3d) [reply Gems/ScriptCanvas/Code/Editor/Assets/ScriptCanvasFileHandling.cpp:94]: Enumerating from the serialize context is very slow

**Comment by carlitosan** [comment]
Closes https://github.com/o3de/o3de/issues/8036
Closes https://github.com/o3de/o3de/issues/8035
Closes https://github.com/o3de/o3de/issues/8003
Closes https://github.com/o3de/o3de/issues/7528

**Pull request merged**